src/entities/inline-fragments/specs/InlineTree.integration.spec.ts

```diff
@@ -0,0 +1,373 @@
+
+      tree.insertText(text, 0);
+
+      expect(tree.getText())
+        .toBe(text + initialText);
+    });
+
+    it('should insert text at the middle of the non-empty tree', () => {
+      const initialText = 'Editor.js is a block-styled editor.';
+      const index = 10;
+      const child = new TextNode({ value: initialText });
+
+      const tree = new RootInlineNode({ children: [ child ] });
+      const text = ' Editor outputs clean data in JSON ';
+
+      tree.insertText(text, index);
+
+      expect(tree.getText())
+        .toBe(initialText.slice(0, index) + text + initialText.slice(index));
+    });
+  });
+
+  describe('text removal', () => {
+    it('should return removed text', () => {
+      const initialText = 'Editor.js is a block-styled editor.';
+      const child = new TextNode({ value: initialText });
+      const tree = new RootInlineNode({ children: [ child ] });
+
+      const removedText = tree.removeText();
+
+      expect(removedText)
+        .toBe(initialText);
+    });
+
+    it('should remove all text from the tree', () => {
+      const initialText = 'Editor.js is a block-styled editor.';
+      const child = new TextNode({ value: initialText });
+      const tree = new RootInlineNode({ children: [ child ] });
+
+      tree.removeText();
+
+      expect(tree.length)
+        .toBe(0);
+    });
+
+    it('should remove text from the beginning of the tree', () => {
+      const initialText = 'Editor.js is a block-styled editor.';
+      const child = new TextNode({ value: initialText });
+      const tree = new RootInlineNode({ children: [ child ] });
+      const index = 10;
+
+      tree.removeText(0, index);
+
+      expect(tree.getText())
+        .toBe(initialText.slice(index));
+    });
+
+    it('should remove text from the middle of the tree', () => {
+      const initialText = 'Editor.js is a block-styled editor.';
+      const child = new TextNode({ value: initialText });
+      const tree = new RootInlineNode({ children: [ child ] });
+      const index = 10;
+      const length = 5;
+
+      tree.removeText(index, index + length);
+
+      expect(tree.getText())
+        .toBe(initialText.slice(0, index) + initialText.slice(index + length));
+    });
+
+    it('should remove text from the end of the tree', () => {
+      const initialText = 'Editor.js is a block-styled editor.';
+      const child = new TextNode({ value: initialText });
+      const tree = new RootInlineNode({ children: [ child ] });
+      const index = 10;
+
+      tree.removeText(index);
+
+      expect(tree.getText())
+        .toBe(initialText.slice(0, index));
+    });
+
+    it('should throw an error if index is out of range', () => {
+      const initialText = 'Editor.js is a block-styled editor.';
+      const child = new TextNode({ value: initialText });
+      const tree = new RootInlineNode({ children: [ child ] });
+      const index = 100;
+
+      expect(() => tree.removeText(index))
+        .toThrowError();
```

Comment:
could we also check the error text or type here?

Reply:
It's not a good idea to check for error text as we might change it later. We can use custom error classes, but that's a whole big task

Reply:
If we change the error text, we can also change it in tests. For now any error thrown during execution of `removeText()` will lead false-positive test assertion